### ISSUE TITLE
ios_snmp_server, facts failing.

### ISSUE THREAD

**Issue by tin-ot** [ansible-collections/cisco.ios#495]
##### SUMMARY

Impossible to execute the module.

maybe related to  SNMP view fact collection failure #491 

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
ansible [core 2.11.7] 
```

##### COLLECTION VERSION

```paste below
cisoc.ios 2.6.0
```

##### CONFIGURATION

```paste below
Can be provided if relevant
```

##### OS / ENVIRONMENT

controller : 
rhel 8.4

Switch :

Model Number                       : C9300-48UXM

Switch Ports Model              SW Version        SW Image              Mode
------ ----- -----              ----------        ----------            ----
*    1 64    C9300-48UXM        16.9.5            CAT9K_IOSXE           INSTALL
     2 64    C9300-48UXM        16.9.5            CAT9K_IOSXE           INSTALL

##### STEPS TO REPRODUCE

```yaml
        - name: Replace commands with provided configuration
          cisco.ios.ios_snmp_server:
            config:
              location: 'updated location entry'
              packet_size: 500
              communities:
                - acl_v4: acl_uq
                  name: communityOverriden
                  rw: true
            state: replaced

```
or any-other configuration.
Fails with state: gathered as well.

##### EXPECTED RESULTS

module executes on device.

##### ACTUAL RESULTS

```paste below
The full traceback is:
  File "/tmp/ansible_cisco.ios.ios_snmp_server_payload_3vbhkc89/ansible_cisco.ios.ios_snmp_server_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/facts/facts.py", line 134, in get_network_resources_facts
    self._connection, self.ansible_facts, data
  File "/tmp/ansible_cisco.ios.ios_snmp_server_payload_3vbhkc89/ansible_cisco.ios.ios_snmp_server_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py", line 87, in populate_facts
    self.sort_list_dicts(objs)
  File "/tmp/ansible_cisco.ios.ios_snmp_server_payload_3vbhkc89/ansible_cisco.ios.ios_snmp_server_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py", line 52, in sort_list_dicts
    objs[k] = sorted(objs[k], key=lambda _k: str(_k[p_key[k]]))
  File "/tmp/ansible_cisco.ios.ios_snmp_server_payload_3vbhkc89/ansible_cisco.ios.ios_snmp_server_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py", line 52, in <lambda>
    objs[k] = sorted(objs[k], key=lambda _k: str(_k[p_key[k]]))
fatal: [host]: FAILED! => {"changed": false, "msg": "'username'"}
```

**Comment by KB-perByte** [comment]
Hey @tin-ot,
I am not sure as you might be facing an issue with the `view` attribute not working correctly, as you have rightly tagged the open issue there is a fix for that waiting to get merged,
Without any initial config for snmp the module works fine on my end, sharing the logs for the above-mentioned use case.
Play exec -
```
changed: [mrIos] => {
    "after": {
        "communities": [
            {
                "acl_v4": "acl_uq",
                "name": "communityOverriden",
                "rw": true
            }
        ],
        "location": "updated location entry",
        "packet_size": 500
    },
    "before": {},
    "changed": true,
    "commands": [
        "snmp-server location updated location entry",
        "snmp-server packetsize 500",
        "snmp-server community communityOverriden rw acl_uq"
    ],
    "invocation": {
        "module_args": {
            "config": {
                "accounting": null,
                "cache": null,
                "chassis_id": null,
                "communities": [
                    {
                        "acl_v4": "acl_uq",
                        "acl_v6": null,
                        "name": "communityOverriden",
                        "ro": null,
                        "rw": true,
                        "view": null
                    }
                ],
                "contact": null,
                "context": null,
                "drop": null,
                "engine_id": null,
                "file_transfer": null,
                "groups": null,
                "hosts": null,
                "if_index": null,
                "inform": null,
                "ip": null,
                "location": "updated location entry",
                "manager": null,
                "packet_size": 500,
                "password_policy": null,
                "queue_length": null,
                "source_interface": null,
                "system_shutdown": null,
                "trap_source": null,
                "trap_timeout": null,
                "traps": null,
                "users": null,
                "views": null
            },
            "running_config": null,
            "state": "replaced"
        }
    }
}
```
gathered state - 
```
ok: [localhost] => {
    "changed": false,
    "gathered": {
        "communities": [
            {
                "acl_v4": "acl_uq",
                "name": "communityOverriden",
                "rw": true
            }
        ],
        "location": "updated location entry",
        "packet_size": 500
    },
    "invocation": {
        "module_args": {
            "config": null,
            "running_config": null,
            "state": "gathered"
        }
    }
}
```
Confirming, Is the issue that you are facing residing among the set of config you configured by the above play?
as the logs say it reached to process `username` please wait for the fix for https://github.com/ansible-collections/cisco.ios/issues/491 to get merged.
Thanks

**Comment by KB-perByte** [comment]
Hey @tin-ot, tagging this issue with fix #492.
Thanks
